src/goto-analyzer/show_callsites.cpp

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************\
+
+Module: Write a function call graph to a CSV file
+
+Author: Norbert Manthey, nmanthey@amazon.de
+
+\*******************************************************************/
+
+#include "show_callsites.h"
+
+#include <goto-programs/compute_called_functions.h>
+#include <goto-programs/goto_model.h>
+#include <util/cprover_prefix.h>
+#include <util/json.h>
+#include <util/prefix.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+/** show_callsites writes a call graph to CSV
+ *
+ * This function creates a map between functions and consumed functions, and
+ * stores this relationship in a CSV file, to be processed outside of
+ * goto-analyzer lateron.
+ *
+ * Functions, whose address is consumed, are listed in a separate section so
+ * that outside reasoning has visibility as well.
+ **/
+
+void show_callsites(
+  const goto_modelt &goto_model,
+  const std::string &output_filename)
+{
+  // generate all functions which we cannot be sure where they are called
+  // because their address is consumed, and stored into function pointers
+  std::unordered_set<irep_idt> excluded_functions;
+  compute_address_taken_functions(
+    goto_model.goto_functions, excluded_functions);
+
+  std::ofstream output_file;
+
+  if(output_filename != "-")
+  {
+    output_file.open(output_filename);
+    if(!output_file.is_open())
+      throw "failed to open callsite output file";
+  }
+
+  std::ostream *output_stream =
+    output_filename == "-" ? &std::cout : &output_file;
+
+  // handle functions whose address is used
+  json_objectt result;
+
+  json_arrayt by_address;
+  for(const auto &name : excluded_functions)
+  {
+    const std::string str_name = id2string(name);
+    if(
+      has_prefix(str_name, CPROVER_PREFIX) ||
+      (str_name.find('$') != std::string::npos))
```

Comment:
Unnecessary braces, I'd think that everyone knows order of operators. Also, since you're checking this twice, you could put it into a static function. That would reduce the size of the if statement, so then you wouldn't be forced to wrap `continue` in braces, which in the end means - less lines!